lib/features/steps/feature_flag_steps.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# @!group Feature flag steps
+
+# Verifies that the are no feature flags present
+Then('the event has no feature flags') do
+  featureFlags = Maze::Helper.read_key_path(Maze::Server.errors.current[:body], "events.0.featureFlags")
+  assert(featureFlags.nil? || featureFlags.empty?, "Feature flags expected to be nil or empty, was #{featureFlags}")
+end
+
+# Verifies a feature flag with a specific variant is uniquely present
+#
+# @step_input flag_name [String] The featureFlag value expected
+# @step_input variant [String] The variant value expected
+Then('the event contains the feature flag {string} with variant {string}') do |flag_name, variant|
+  featureFlags = Maze::Helper.read_key_path(Maze::Server.errors.current[:body], "events.0.featureFlags")
+  assert_false(featureFlags.nil? || featureFlags.empty?, "Feature flags were nil or empty: #{featureFlags}")
+  assert(featureFlags.one? { |flag| flag['featureFlag'] == flag_name && flag['variant'] == variant })
```

Comment:
Can `variant` ever be anything other than a string? I think Cucumber will always provide a string when matching `{string}`, but I'm not 100% sure

e.g. if `flag['variant']` was an integer `123`, this should fail:

```gherkin
Then the event contains the feature flag "name" with variant "123"
```

Reply:
I read in the ED that variant should always be assumed to be a string, is this not the case?

Reply:
Yes, sorry that was what I was trying to say — it should be impossible for the `variant` variable to be anything other than a string, otherwise the `==` could pass e.g. if they were both integers

I think it's fine as-is, just wanted to check how cucumber handles `{string}` with values that could be considered non-strings (`123`/`null`/etc&hellip;)

Reply:
Cucumber is type sensitive, e.g. for :
```
{
  "values": {
    "string":"123",
    "integer":123"
  }
}

This both pass:
```
    And the error payload field "values.string" equals "123"
    And the error payload field "values.integer" equals 123
```

and these both fail:
```
    And the error payload field "values.string" equals 123
    And the error payload field "values.integer" equals "123"
```

Reply:
...so the fact that the step uses `variant {string}` and not `variant {int}` means it's ok.